front/lib/api/assistant/visibility.ts

```diff
@@ -0,0 +1,76 @@
+import { getAgentConfiguration } from "@app/lib/api/assistant/configuration";
+import { Authenticator } from "@app/lib/auth";
+import { Membership } from "@app/lib/models";
+import { MemberAgentVisibility } from "@app/lib/models/assistant/agent";
+import { Err, Ok, Result } from "@app/lib/result";
+import { MemberAgentVisibilityType } from "@app/types/assistant/agent";
+
+export async function upsertAgentVisibility({
+  auth,
+  agentId,
+  visibility,
```

Comment:
Obviously this would mean have an internal DB type and one that is external which is just the visibility for the user?

Reply:
So to make clear on your proposal. The current situation
- user wants "visible" <-> 'published listed' in the table => set that = call to upsert 'published-listed'
- user wants "hidden" <-> 'workspace-unlisted' in the table => set that = call to upsert 'workspace-unlisted'
- default behaviour (shown for workspace, hidden for ) <-> no row in the table => set that = call to delete
The contents of the table are independent from the scope, e.g. you can have a 'published' scope agent with users having for that agent a 'worskpace-unlisted' entry. This would correspond e.g. to the case where an agent was elevated to workspace (visible by all user), and user X removed it from their list. Then, the agent was delevated (back to 'published' scope). So user X still has that agent in 'workspace-unlisted', meaning if the agent becomes workspace again, they will still not have it in their list as per their initial wish.

Reply:
You are suggesting a front type that says visible / hidden / default, and writes the correct thing in the table

Reply:
Not sure it's needed but still thinking
but maybe we could actually just rename 'workspace-unlisted' to 'hidden' and 'published-listed' to 'visible'

Reply:
No strong opinion. Would go for no-op since simpler change but ok for both, let's tranche IRL when you've read that